File: daiphat-fe/src/admin/constants/permission.constants.ts
/**
 * ALL PERMISSION CODES
 * Limited to essential core modules currently existing in Backend AppPermission.java
 */
export const PERMISSIONS = {
    DASHBOARD: {
        SYSTEM: "dashboard:system",
        ANALYTICS: "dashboard:analytics",
        ECOMMERCE: "dashboard:ecommerce",
    },
    STATISTICS: {
        REVENUE: "stats:revenue",
        ORDER: "stats:order",
        SERVICE: "stats:service",
    },
    USER: {
        VIEW: "member:view",
        CREATE: "member:create",
        EDIT: "member:edit",
        DELETE: "member:delete",
    },
    ACCOUNT: {
        VIEW: "admin:view",
        CREATE: "admin:create",
        EDIT: "admin:edit",
        DELETE: "admin:delete",
    },
    STREET_AGENT: {
        VIEW: "streetAgent:view",
        CREATE: "streetAgent:create",
        EDIT: "streetAgent:edit",
        DELETE: "streetAgent:delete",
    },
    ROLE: {
        VIEW: "role:view",
        CREATE: "role:create",
        EDIT: "role:edit",
        DELETE: "role:delete",
    },
    ARTICLE: {
        VIEW: "article:view",
        CREATE: "article:create",
        EDIT: "article:edit",
        DELETE: "article:delete",
    },
    TICKET: {
        VIEW: "ticket:view",
        CREATE: "ticket:create",
        EDIT: "ticket:edit",
        DELETE: "ticket:delete",
    },
    IMPORT_BATCH: {
        VIEW: "importBatch:view",
        CREATE: "importBatch:create",
    },
    SUPPLIER: {
        VIEW: "supplier:view",
        CREATE: "supplier:create",
        EDIT: "supplier:edit",
    },
    PROVIDER: {
        VIEW: "provider:view",
        CREATE: "provider:create",
        EDIT: "provider:edit",
        DELETE: "provider:delete",
        SYNC: "provider:sync",
    },
    TICKET_SERVICE: {
        VIEW: "ticketService:view",
        CREATE: "ticketService:create",
        EDIT: "ticketService:edit",
    },
    ORDER: {
        VIEW: "order:view",
        CREATE: "order:create",
        EDIT: "order:edit",
        DELETE: "order:delete",
    },
    REFUND: {
        VIEW: "refund:view",
        PROCESS: "refund:process",
    },
    PRIZE_PAYOUT: {
        VIEW: "prizePayout:view",
        PROCESS: "prizePayout:process",
    },
    REGION: {
        VIEW: "region:view",
        CREATE: "region:create",
        EDIT: "region:edit",
        DELETE: "region:delete",
        SYNC: "region:sync",
    },
    PRIZE_STRUCTURE: {
        VIEW: "prizeStructure:view",
        CREATE: "prizeStructure:create",
        EDIT: "prizeStructure:edit",
        DELETE: "prizeStructure:delete",
        SYNC: "prizeStructure:sync",
    },
    LOTTERY_RESULT: {
        VIEW: "lotteryResult:view",
        CREATE: "lotteryResult:create",
        EDIT: "lotteryResult:edit",
        DELETE: "lotteryResult:delete",
        SYNC: "lotteryResult:sync",
    },
    TICKET_SERVICE_ORDER: {
        VIEW: "ticketServiceOrder:view",
        EDIT: "ticketServiceOrder:edit",
    },
    CHAT: {
        VIEW: "chat:view",
        MANAGE: "chat:manage",
    },
    NOTIFICATION: {
        VIEW: "notification:view",
    },
    REVIEW: {
        VIEW: "review:view",
    },
    SUPPORT_TICKET: {
        VIEW: "supportTicket:view",
        PROCESS: "supportTicket:process",
    },
    CALENDAR: {
        VIEW: "calendar:view",
    },
    SETTINGS: {
        VIEW: "settings:view",
        EDIT: "settings:edit",
    },
} as const;

File: daiphat-be/core-api/src/main/java/com/daiphat/coreapi/domain/model/enums/auth/PermissionConstants.java
package com.daiphat.coreapi.domain.model.enums.auth;

import lombok.AccessLevel;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
public final class PermissionConstants {

    // --- RESOURCES ---
    public static final String DASHBOARD = "dashboard";
    public static final String STATISTICS = "stats";
    public static final String USER = "member";
    public static final String ACCOUNT = "admin";
    public static final String ROLE = "role";
    public static final String ARTICLE = "article";
    public static final String TICKET = "ticket";
    public static final String IMPORT_BATCH = "importBatch";
    public static final String SUPPLIER = "supplier";
    public static final String STATION = "station";
    public static final String REGION = "region";
    public static final String PRIZE_STRUCTURE = "prizeStructure";
    public static final String LOTTERY_RESULT = "lotteryResult";
    public static final String PROVIDER = "provider";
    public static final String STREET_AGENT = "streetAgent";
    public static final String ORDER = "order";
    public static final String REFUND = "refund";
    public static final String TICKET_SERVICE = "ticketService";
    public static final String TICKET_SERVICE_ORDER = "ticketServiceOrder";
    public static final String COUPON = "coupon";
    public static final String CHAT = "chat";
    public static final String CALENDAR = "calendar";
    public static final String SETTINGS = "settings";

    // --- ACTIONS ---
    public static final String VIEW = ":view";
    public static final String CREATE = ":create";
    public static final String EDIT = ":edit";
    public static final String DELETE = ":delete";
    public static final String MANAGE = ":manage";
    public static final String SYNC = ":sync";
    public static final String SYSTEM = ":system";
    public static final String ANALYTICS = ":analytics";
    public static final String ECOMMERCE = ":ecommerce";
    public static final String APPROVE = ":approve";
    public static final String REJECT = ":reject";
    public static final String PROCESS = ":process";
}
